TrackerMC/src/StrawDigisFromStrawGasSteps_module.cc

```diff
@@ -1225,28 +1237,47 @@ namespace mu2e {
       _sdiag->Fill();
     }
 
-    StrawPosition StrawDigisFromStrawGasSteps::strawPosition( XYZVectorF const& cpos,Straw const& straw) const {
+    StrawCoordinates StrawDigisFromStrawGasSteps::strawCoordinates( XYZVectorF const& cpos, Straw const& straw) const {
+      StrawCoordinates retval;
       static XYZVectorF zdir(0.0,0.0,1.0);
-      XYZVectorF smid = XYZVectorF(straw.getMidPoint());
-      XYZVectorF delta = cpos - smid; // cluster position WRT straw middle
-      XYZVectorF sdir = XYZVectorF(straw.getDirection());
+      XYZVectorF smid = XYZVectorF(straw.strawPosition());
+      XYZVectorF delta = cpos - smid; // cluster position WRT nominal straw middle
+      XYZVectorF sdir = XYZVectorF(straw.strawDirection());
       XYZVectorF pdir = sdir.Cross(zdir); // radial direction
       if(pdir.Dot(smid) < 0.0)pdir *= -1.0; // sign radially outwards
       float dw = delta.Dot(sdir);
       XYZVectorF cperp = delta - dw*sdir; // just perp part
-      float phi = atan2(cperp.Dot(pdir),cperp.Dot(zdir));// angle around wire WRT Z axis in range -pi,pi
+      float phi = atan2(cperp.Dot(pdir),cperp.Dot(zdir)); // angle around wire WRT z axis in range -pi,pi
       float rho = min(sqrt(cperp.mag2()),(float)_rstraw); // truncate!
-      return StrawPosition(rho,dw,phi);
+      retval._strawPosition = StrawPosition(rho,dw,phi);
+      if (_simWireOffsets){
```

Comment:
As remarked above, I would perefer we drive this behavior purely on database contents, ie always correct for wire offsets, but if the database purpose is 'perfect' those will be 0.